apiserver/facades/client/bundle/bundle.go

```diff
@@ -103,6 +103,62 @@ func (b *BundleAPI) checkCanRead() error {
 	return nil
 }
 
+// GetChanges returns the list of changes required to deploy the given bundle
+// data. The changes are sorted by requirements, so that they can be applied in
+// order.
+// V1 GetChanges did not support device.
+func (b *APIv1) GetChanges(args params.BundleChangesParams) (params.BundleChangesResults, error) {
```

Comment:
so is this pointer to a APIv1 or APIv2?
It feels very similar to what we have done in modelmanager...
For example see https://github.com/juju/juju/blob/develop/apiserver/facades/client/modelmanager/modelmanager.go#L835 vs https://github.com/juju/juju/blob/develop/apiserver/facades/client/modelmanager/modelmanager.go#L854.
This is an example of a method change in between different versions. 
You will need to keep an APIv1 call to return data without device support and have another call sitting on APIv2 providing the support. Internally they may call the same func but present data differently.

Reply:
yes, agreed. I did think this. But here the difference between two versions are neither params nor returns. It's in the middle `GUIArgs()`. The idea solution would do whole versioning in bundlechanges repo. I was thinking to do it like that. But I am not very sure if it's worth to put this effect.

Reply:
Really? It looks like a brand new method to me?...